lib/http/connection.rb

```diff
@@ -93,7 +93,7 @@ def readpartial(size = BUFFER_SIZE)
       chunk    = @parser.read(size)
       finish_response if finished
 
-      chunk.to_s
+      chunk || "".b
```

Comment:
Hmm, looks  bit weird to me - why isn't it manages encoding of `chunk` if it's not `nil`? o_O

Reply:
Oh. commit message clears things up. All good, I guess.